### DISCUSSION TITLE
'Token exchange' support by 'keycloak-js adapter'

### DISCUSSION THREAD

**Discussion by antonfil** [keycloak/keycloak#17018] [Q&A]
Is there a plan to add 'Token exchange' (https://www.keycloak.org/docs/latest/securing_apps/#_token-exchange) support to 'keycloak-js adapter' (https://www.keycloak.org/docs/latest/securing_apps/index.html#_javascript_adapter)?

**Comment by jonkoops** [comment]
Not at this time no. Development on Keyloak JS has kind of stalled at the moment due to other priorities. In the meantime you might be better of using a 3rd party OpenID Connect client.

**Reply by antonfil** [reply]
@jonkoops could you please suggest a third party JS OpenID Connect client which supports ‘token exchange’ and will work fine with keycloak?

**Reply by jonkoops** [reply]
Any that properly implement the specification should work.
